amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -59,67 +63,106 @@ class SearchPage extends React.Component<SearchPageProps, SearchPageState> {
     },
     users: {
       page_index: 0,
-      // TODO - cleanup sample data
-      results: [{
-        type: ResourceType.user,
-        active: true,
-        birthday: '10-10-1990',
-        department: "Department",
-        email: "mail@email.com",
-        first_name: "Ash",
-        github_username: "Github User",
-        id: 12345,
-        last_name: "Ketchum",
-        manager_email: "manager@company.com",
-        name: "Ash Ketchum",
-        offboarded: false,
-        office: "city",
-        role: "Pokemon Trainer",
-        start_date: "10-10-2014",
-        team_name: "Team Red",
-        title: "Pikachu",
-      },{
-        type: ResourceType.user,
-        active: false,
-        birthday: '10-10-1990',
-        department: "Department",
-        email: "mail@email.com",
-        first_name: "Gary",
-        github_username: "Github User",
-        id: 12345,
-        last_name: "Oak",
-        manager_email: "manager@company.com",
-        name: "Gary Oak",
-        offboarded: false,
-        office: "city",
-        role: "Pokemon Trainer",
-        start_date: "10-10-2014",
-        team_name: "Team Blue",
-        title: "Eevee",
-      }],
-      total_results: 1,
+      results: [],
+      total_results: 0,
     }
   };
 
   constructor(props) {
     super(props);
 
-    this.handlePageChange = this.handlePageChange.bind(this);
-    this.updateQueryString = this.updateQueryString.bind(this);
+    this.state = {
+      selectedTab: ResourceType.table,
+    };
   }
 
   componentDidMount() {
     this.props.getPopularTables();
 
     const params = qs.parse(window.location.search);
-    const searchTerm = params['searchTerm'];
-    const pageIndex = params['pageIndex'];
+    const { searchTerm, pageIndex, selectedTab} = params;
+
+    this.updateSelectedTab(selectedTab);
     if (searchTerm && searchTerm.length > 0) {
       const index = pageIndex || '0';
-      this.props.executeSearch(searchTerm, index);
+      this.props.searchAll(searchTerm, this.getSearchOptions(index));
```

Comment:
This method relies on the value of `this.state.selectedTab`, which is reset a couple lines up in `updateSelectedTab`. There is no guarantee that by the time we call `getSearchOptions` that the component will have completed whatever black-box logic is involved in `setState` -- it's possible for `this.state.selectedTab` to not yet have been updated to what you intend it to be.  I would suggest explicitly passing the  intended `selectedTab` into `getSearchOptions`.

Reply:
I see what you mean. I'll create a separate 'validateTab' method and then use whatever value gets passed back from that as a valid tab.